kupyna/src/utils.rs

```diff
@@ -16,11 +18,32 @@ fn multiply_gf(mut x: u8, mut y: u8) -> u8 {
             x ^= REDUCTION_POLYNOMIAL as u8;
         }
         y >>= 1;
+        i += 1;
     }
 
     r
 }
 
+const fn generate_gf_lookup_table() -> [[u8; 256]; 256] {
+    let mut table = [[0u8; 256]; 256];
+    let mut x = 0;
+    while x < 256 {
+        let mut y = 0;
+        while y < 256 {
+            table[x][y] = gf_multiply(x as u8, y as u8);
+            y += 1;
+        }
+        x += 1;
+    }
+    table
+}
+
+pub(crate) static GF_LOOKUP_TABLE: [[u8; 256]; 256] = generate_gf_lookup_table();
```

Comment:
I think you can keep this static private since it's only used by the `multiply_gf` function.